lib/route53.js

```diff
@@ -28,53 +74,23 @@ class AWSHandler extends DNSHandler {
 		return this._awsSdk;
 	}
 
-	_getHostedZoneId() {
+	getHostedZoneId() {
 		return this.Route53.listHostedZonesByName({
 			DNSName: this.domain,
 		}).promise()
-			.then(response => _.filter(_.flatMap(response), [
-				"Config.PrivateZone",
-				false,
-			])[0].Id)
-			.catch(error => {
-				console.log("[ERROR]", error);
-			});
-	}
-
-	_generateParams(hostedZoneId, opts) {
-		let params = {
-			HostedZoneId: hostedZoneId,
-			ChangeBatch: {
-				Changes: [
-					{
-						Action: "",
-						ResourceRecordSet: {
-							Name: opts.name.concat(".", this.domain, "."),
-							Type: opts.type || "CNAME",
-							TTL: opts.ttl || 3600,
-						},
-					},
-				],
-			},
-		};
-
-		if (opts.values) {
-			params.ChangeBatch.Changes[0].ResourceRecordSet.ResourceRecords = opts.values.map(function(r){
-				return {
-					Value: r
-				}
-			});
-		}
-		return Promise.resolve(params);
+		.then(response => _.filter(_.flatMap(response), [
+			"Config.PrivateZone",
+			false,
+		])[0].Id)
+		.catch(error => {
+			console.log("[ERROR]", error);
+		});
 	}
 
 	addRecord(opts) {
-		return this._getHostedZoneId()
-			.then(hostedZoneId => this._generateParams(hostedZoneId, opts))
-			.then((params) => {
-				params.ChangeBatch.Changes[0].Action = "UPSERT";
-				return Promise.resolve(params);
-			})
+		return this.getHostedZoneId()
+			.then(hostedZoneId => generateParams(hostedZoneId, this.domain, opts))
+			.then(params => changeRequestMethod(params, "UPSERT"))
 			.then(params => this.Route53.changeResourceRecordSets(params).promise())
 			.then(response => {
 				console.log("[INFO]", "Succesfully updated DNS for " + opts.name);
```

Comment:
Try to always return in a `then()` just in case someone else wants to chain (for whatever reason).